ssv/runner_validations.go

```diff
@@ -97,21 +81,25 @@ func (b *BaseRunner) validateDecidedConsensusData(runner Runner, val types.Encod
 	return nil
 }
 
-func (b *BaseRunner) verifyExpectedRoot(runner Runner, psigMsgs *types.PartialSignatureMessages, expectedRootObjs []ssz.HashRoot, domain spec.DomainType) error {
+func (b *BaseRunner) verifyExpectedRoot(runner Runner, psigMsgs *types.PartialSignatureMessages, expectedRootObjs []ssz.HashRoot, domains []spec.DomainType) error {
 	if len(expectedRootObjs) != len(psigMsgs.Messages) {
 		return errors.New("wrong expected roots count")
 	}
+	if len(expectedRootObjs) != len(domains) {
+		return errors.New("number of expected root differs from number of domains")
+	}
 
 	// convert expected roots to map and mark unique roots when verified
 	sortedExpectedRoots, err := func(expectedRootObjs []ssz.HashRoot) ([][32]byte, error) {
 		epoch := b.BeaconNetwork.EstimatedEpochAtSlot(b.State.StartingDuty.DutySlot())
-		d, err := runner.GetBeaconNode().DomainData(epoch, domain)
-		if err != nil {
-			return nil, errors.Wrap(err, "could not get pre consensus root domain")
-		}
 
 		ret := make([][32]byte, 0)
-		for _, rootI := range expectedRootObjs {
+		for i, rootI := range expectedRootObjs {
+			d, err := runner.GetBeaconNode().DomainData(epoch, domains[i])
+			if err != nil {
+				return nil, errors.Wrap(err, "could not get pre consensus root domain")
```

Comment:
pre consensus?

Reply:
Well seen. Btw, we always had this error string. But I think we can't reach this part in a test.

Reply:
Fixed.